.github/workflows/check_tests.yml

```diff
@@ -5,400 +5,274 @@ on:
-          aws-image-search-pattern: ${{ env.AWS_IMAGE_SEARCH_PATTERN }}
-          aws-image-search-owners: ${{ env.AWS_IMAGE_SEARCH_OWNERS }}
       - uses: actions/checkout@v2
       - name: install sccache
         env:
           SCCACHE_RELEASE_URL: https://github.com/mozilla/sccache/releases/download
           SCCACHE_VERSION: v0.5.3
         run: |
+          apt update
+          apt install -y curl wget nodejs
+          which node
+          which npm
+          npm install -g yarn
           SCCACHE_FILE=sccache-$SCCACHE_VERSION-x86_64-unknown-linux-musl
           mkdir -p $HOME/.local/bin
           curl -L "$SCCACHE_RELEASE_URL/$SCCACHE_VERSION/$SCCACHE_FILE.tar.gz" | tar xz
           mv -f $SCCACHE_FILE/sccache $HOME/.local/bin/sccache
           chmod +x $HOME/.local/bin/sccache
           echo "$HOME/.local/bin" >> $GITHUB_PATH
           echo "$HOME/.cargo/bin" >> $GITHUB_PATH
-      - name: cache cargo registry
-        uses: actions/cache@v2
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: |
-            cargo-
       - name: cache sccache
         uses: actions/cache@v2
         continue-on-error: false
         with:
-          path: /home/runner/.cache/sccache
+          path: ~/.cache/sccache
           key: sccache-${{ hashFiles('**/Cargo.lock') }}
           restore-keys: |
             sccache-
       - name: start sccache server
         run: sccache --start-server
-      - name: build benchmarking binary
+      - name: rust setup
+        shell: bash
         env:
           RUST_BACKTRACE: full
           RUSTC_WRAPPER: sccache
-          SCCACHE_CACHE_SIZE: 120G
-          SCCACHE_DIR: /home/runner/.cache/sccache
+          # SCCACHE_CACHE_SIZE: 120G
+          SCCACHE_CACHE_SIZE: 2G
+          SCCACHE_DIR: ~/.cache/sccache
           CARGO_TERM_COLOR: always
         run: |
-          sudo apt update
-          sudo apt install -y pkg-config libssl-dev protobuf-compiler
+          apt update
+          apt install -y pkg-config libssl-dev protobuf-compiler curl build-essential clang git
           protoc --version
           curl -s https://sh.rustup.rs -sSf | sh -s -- -y
           source ${HOME}/.cargo/env
           rustup toolchain install nightly-2023-03-03
           rustup default nightly-2023-03-03
           rustup target add wasm32-unknown-unknown
-          df -h
-          sudo apt remove -y '^dotnet-.*'
-          sudo apt remove -y '^mongodb-.*'
-          sudo apt remove -y '^mysql-.*'
-          sudo apt remove -y '^postgresql-*'
-          sudo apt remove -y google-cloud-sdk google-chrome-stable firefox powershell mono-devel libgl1-mesa-dri
-          sudo apt autoremove -y
-          sudo apt clean
-          sudo rm -rf /usr/local/lib/android
-          sudo rm -rf /usr/share/dotnet
-          df -h
+      - name: cache cargo
+        uses: Swatinem/rust-cache@v2
```

Comment:
so what exactly is the difference between sccache and this ?

Reply:
yep this caches rust packages
* ~/.cargo
* ./target

aptos using this too :) https://github.com/aptos-labs/aptos-core/blob/main/.github/actions/rust-setup/action.yaml